mistletoe/core_tokens.py

```diff
@@ -1,18 +1,17 @@
 import re
-
+import unicodedata
 
 whitespace = {' ', '\t', '\n', '\x0b', '\x0c', '\r'}
-unicode_whitespace = {'\t', '\n', '\x0b', '\x0c', '\r', '\x1c', '\x1d', '\x1e',
-        '\x1f', ' ', '\x85', '\xa0', '\u1680', '\u2000', '\u2001', '\u2002',
-        '\u2003', '\u2004', '\u2005', '\u2006', '\u2007', '\u2008', '\u2009',
-        '\u200a', '\u2028', '\u2029', '\u202f', '\u205f', '\u3000'}
-punctuation = {'!', '"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',',
```

Comment:
> spec:
>
> A [Unicode punctuation character](https://spec.commonmark.org/0.30/#unicode-punctuation-character) is an [ASCII punctuation character](https://spec.commonmark.org/0.30/#ascii-punctuation-character) or anything in the general Unicode categories Pc, Pd, Pe, Pf, Pi, Po, or Ps.

I didn't add more punctuations to this set due to the following reasons:
1. In the future, there might be new characters in the Unicode categories above. I think it is better for use to use the `unicodedata` library so that we don't change this definition frequently.
2. `unicodedata` doesn't provide a way to get all characters in a specific Unicode category, so, I wrote `is_*` functions instead.
3. Usage of `is_*` functions are more efficient; rather than listing all the things out, we just need to check the given character is a punctuation/whitespace or not.